services/src/main/java/org/apache/druid/cli/CliPeon.java

```diff
@@ -363,7 +372,7 @@ public String getTaskIDFromTask(final Task task)
         new IndexingServiceTuningConfigModule(),
         new InputSourceModule(),
         new ChatHandlerServerModule(properties),
-        new LookupModule()
+        LookupLoadingMode.NONE == loadLookups ? new LookupSerdeModule() : new LookupModule()
```

Comment:
Maybe put this logic in a private method `getLookupModule()` and add a javadoc there explaining what we are doing.

Reply:
A log line which indicates the chosen lookup loading mode and the expected effects would also be helpful.

Reply:
added log and private method. also added tests for it